### PR TITLE
powerman role: powerman.conf file contains too many blank lines

### PR DESCRIPTION
**Pull request by MarbolanGos**
Hi,

In powerman role, the template powerman.conf.j2 generates a lot of blank lines into the /etc/powerman/powerman.conf file.
Please see the PR dealing with this allowing a better reading of the extracted file.

### CODE REVIEW

**Comment by oxedions** [comment]
Dear @MarbolanGos 

Thanks a lot for this PR. I will have a look as soon as possible 😊

File: roles/addons/powerman/templates/powerman.conf.j2 [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 include "/etc/powerman/ipmipower.dev"
 
-{% if icebergs_system == true %}
+{%- if icebergs_system == true %}
```
Comment (btravouillon) [inline roles/addons/powerman/templates/powerman.conf.j2:7]: This `-` is not needed.

File: roles/addons/powerman/templates/powerman.conf.j2 [hunk 2]
```diff
@@ -14,20 +14,23 @@ include "/etc/powerman/ipmipower.dev"
   {% endfor %}
 {% else %}
   {% set range = groups['all'] %}
-{% endif %}
+{%- endif %}
```
Comment (btravouillon) [inline roles/addons/powerman/templates/powerman.conf.j2:17]: This `-` is not needed.

File: roles/addons/powerman/templates/powerman.conf.j2 [hunk 3]
```diff
@@ -14,20 +14,23 @@ include "/etc/powerman/ipmipower.dev"
...
-    {% if host in range and hostvars[host]['bmc']['name'] is defined %}
-      {{ hosts.append(host) }}
-      {{ bmcs.append(hostvars[host]['bmc']['name']) }}
-    {% endif %}
-  {% endfor %}
-  {% if hosts | length > 0 %}
+{%- for equipment in j2_equipment_groups_list %}
+  {%- set hosts = [] -%}
+  {%- set bmcs = [] -%}
+  {%- for host in groups[equipment] %}
```
Comment (btravouillon) [inline roles/addons/powerman/templates/powerman.conf.j2:22]: These `-` are not needed.

File: roles/addons/powerman/templates/powerman.conf.j2 [hunk 4]
```diff
@@ -14,20 +14,23 @@ include "/etc/powerman/ipmipower.dev"
...
+    {%- if host in range and hostvars[host]['bmc']['name'] is defined %}
+      {{- hosts.append(host) }}
+      {{- bmcs.append(hostvars[host]['bmc']['name']) }}
+    {%- endif %}
+  {%- endfor %}
+  {%- if hosts | length > 0 %}
```
Comment (btravouillon) [inline roles/addons/powerman/templates/powerman.conf.j2:28]: These `-` are not needed.

File: roles/addons/powerman/templates/powerman.conf.j2 [hunk 5]
```diff
@@ -14,20 +14,23 @@ include "/etc/powerman/ipmipower.dev"
...
+
```
Comment (btravouillon) [inline roles/addons/powerman/templates/powerman.conf.j2:29]: This adds too much blank lines, can you please remove this one?

File: roles/addons/powerman/templates/powerman.conf.j2 [hunk 6]
```diff
@@ -14,20 +14,23 @@ include "/etc/powerman/ipmipower.dev"
...
+# {{ equipment }}
 device "{{ equipment }}" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h {{ bmcs|join(',') }} -u {{ hostvars[hosts[0]]['ep_equipment_authentication']['user'] }} -p {{ hostvars[hosts[0]]['ep_equipment_authentication']['password'] }} |&"
 node "{{ hosts|join(',') }}" "{{ equipment }}" "{{ bmcs|join(',') }}"
-  {% endif %}
-{% endfor %}
+
+  {%- endif %}
+{%- endfor %}
```
Comment (btravouillon) [inline roles/addons/powerman/templates/powerman.conf.j2:35]: These `-` are not needed.

File: roles/addons/powerman/templates/powerman.conf.j2 [hunk 7]
```diff
@@ -14,20 +14,23 @@ include "/etc/powerman/ipmipower.dev"
...
```
Comment (btravouillon) [inline roles/addons/powerman/templates/powerman.conf.j2:36]: Can you please remove this last blank line as well?

**Review by btravouillon — CHANGES_REQUESTED** [review]
A few comments in line. A lot of `-` were added while not needed.
I believe the `if/endif` block at lines 11 and 13 need a `-`.

File: roles/addons/powerman/templates/powerman.conf.j2 (hunk 1 above)
Comment (MarbolanGos) [reply roles/addons/powerman/templates/powerman.conf.j2:7]: I disagree, if I remove this - and the one in line 17 I have a line created:
```
 --- before: /etc/powerman/powerman.conf                                                                                                                                                  +++ after: /root/.ansible/tmp/ansible-local-1925480kd2s2ita/tmp7xxqpibp/powerman.conf.j2                                                                                                 @@ -2,6 +2,8 @@                                                                                                                                                                           ## Ansible managed file, do not edit

 include "/etc/powerman/ipmipower.dev"
+
+
```

File: roles/addons/powerman/templates/powerman.conf.j2 (hunk 2 above)
Comment (MarbolanGos) [reply roles/addons/powerman/templates/powerman.conf.j2:17]: See comment here: https://github.com/bluebanquise/bluebanquise/pull/541#discussion_r608388253

**Comment by MarbolanGos** [comment]
> A few comments in line. A lot of `-` were added while not needed.
> I believe the `if/endif` block at lines 11 and 13 need a `-`.

Agree but not tested on multi iceberg...

File: roles/addons/powerman/templates/powerman.conf.j2 (hunk 3 above)
Comment (MarbolanGos) [reply roles/addons/powerman/templates/powerman.conf.j2:22]: Seems ok to not have them.

File: roles/addons/powerman/templates/powerman.conf.j2 (hunk 4 above)
Comment (MarbolanGos) [reply roles/addons/powerman/templates/powerman.conf.j2:28]: Ok to remove them also.

File: roles/addons/powerman/templates/powerman.conf.j2 (hunk 5 above)
Comment (MarbolanGos) [reply roles/addons/powerman/templates/powerman.conf.j2:29]: This is needed to separate different types of equipments, else it is really difficult to read the file.

File: roles/addons/powerman/templates/powerman.conf.j2 (hunk 6 above)
Comment (MarbolanGos) [reply roles/addons/powerman/templates/powerman.conf.j2:35]: I disagree:

- First dash for endif removed adds two lines between each section
- The dash for ednfor can be removed.

File: roles/addons/powerman/templates/powerman.conf.j2 (hunk 7 above)
Comment (MarbolanGos) [reply roles/addons/powerman/templates/powerman.conf.j2:36]: ok

File: roles/addons/powerman/templates/powerman.conf.j2 (hunk 5 above)
Comment (btravouillon) [reply roles/addons/powerman/templates/powerman.conf.j2:29]: With a blank line at l33, I expect each equipment to be separated from the other with one blank line. Am I wrong?

**Comment by btravouillon** [comment]
With current `master` and simple_cluster inventory:
```
# cat /etc/powerman/powerman.conf 
#### Blue Banquise file ####
## Ansible managed

include "/etc/powerman/ipmipower.dev"


      
      
      
      
      
      
      
      
device "equipment_typeC" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h bc001,bc002,bc003,bc004 -u ADMIN -p ADMIN |&"
node "c001,c002,c003,c004" "equipment_typeC" "bc001,bc002,bc003,bc004"
      
      
device "equipment_typeL" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h blogin1 -u ADMIN -p ADMIN |&"
node "login1" "equipment_typeL" "blogin1"
      
      
device "equipment_typeM" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h bmanagement1 -u ADMIN -p ADMIN |&"
node "management1" "equipment_typeM" "bmanagement1"

```

I do agree this is ugly.

With your change (afc7fc7):
```
# cat /etc/powerman/powerman.conf 
#### Blue Banquise file ####
## Ansible managed

include "/etc/powerman/ipmipower.dev"
# equipment_typeC
device "equipment_typeC" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h bc001,bc002,bc003,bc004 -u ADMIN -p ADMIN |&"
node "c001,c002,c003,c004" "equipment_typeC" "bc001,bc002,bc003,bc004"
# equipment_typeL
device "equipment_typeL" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h blogin1 -u ADMIN -p ADMIN |&"
node "login1" "equipment_typeL" "blogin1"
# equipment_typeM
device "equipment_typeM" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h bmanagement1 -u ADMIN -p ADMIN |&"
node "management1" "equipment_typeM" "bmanagement1"
```

I find this difficult to read and would add some blank lines between each equipment type.

If I apply my recommended changes on top of your last commit (afc7fc7):
```
# cat /etc/powerman/powerman.conf 
#### Blue Banquise file ####
## Ansible managed

include "/etc/powerman/ipmipower.dev"

# equipment_typeC
device "equipment_typeC" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h bc001,bc002,bc003,bc004 -u ADMIN -p ADMIN |&"
node "c001,c002,c003,c004" "equipment_typeC" "bc001,bc002,bc003,bc004"

# equipment_typeL
device "equipment_typeL" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h blogin1 -u ADMIN -p ADMIN |&"
node "login1" "equipment_typeL" "blogin1"

# equipment_typeM
device "equipment_typeM" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h bmanagement1 -u ADMIN -p ADMIN |&"
node "management1" "equipment_typeM" "bmanagement1"

```

The diff:
```diff
diff --git a/roles/addons/powerman/templates/powerman.conf.j2 b/roles/addons/powerman/templates/powerman.conf.j2
index f9e1d51..0749197 100644
--- a/roles/addons/powerman/templates/powerman.conf.j2
+++ b/roles/addons/powerman/templates/powerman.conf.j2
@@ -4,7 +4,7 @@
 
 include "/etc/powerman/ipmipower.dev"
 
-{%- if icebergs_system == true %}
+{% if icebergs_system == true %}
 # Iceberg system is on
   {% set range = groups[j2_current_iceberg] %}
   {% for host in groups[managements_group_name] %}
@@ -14,7 +14,7 @@ include "/etc/powerman/ipmipower.dev"
   {% endfor %}
 {% else %}
   {% set range = groups['all'] %}
-{%- endif %}
+{% endif %}
 
 {%- for equipment in j2_equipment_groups_list %}
   {% set hosts = [] %}
@@ -26,10 +26,9 @@ include "/etc/powerman/ipmipower.dev"
     {%- endif %}
   {% endfor %}
   {% if hosts | length > 0 %}
-
 # {{ equipment }}
 device "{{ equipment }}" "ipmipower" "/usr/sbin/ipmipower --wait-until-on --wait-until-off -h {{ bmcs|join(',') }} -u {{ hostvars[hosts[0]]['ep_equipment_authentication']['user'] }} -p {{ hostvars[hosts[0]]['ep_equipment_authentication']['password'] }} |&"
 node "{{ hosts|join(',') }}" "{{ equipment }}" "{{ bmcs|join(',') }}"
 
-  {%- endif %}
+  {% endif %}
 {% endfor %}
```

**Comment by MarbolanGos** [comment]
I agree your last comment, this version seems to be more human readable. I have commited your patch. I think all is set now! Thanks.

**Review by btravouillon — DISMISSED** [review]
Looks good to me. 👍

**Comment by btravouillon** [comment]
The role moved to roles/core since, you need to rebase your PR on top of `master`.

**Pull request closed without merging**